src/PuppeteerRunnerExtension.ts

```diff
@@ -14,22 +14,22 @@
     limitations under the License.
  */
 
-import { RunnerExtension } from './RunnerExtension.js';
+import { TrivialRunnerExtension } from './TrivialRunnerExtension.js';
```

Comment:
What's the argument for having this TrivialRunnerExtension? The RunnerExtension is a virtual class on purpose so that the sub-classes don't need to check for the existence of `super.HOOK_NAME`.

Reply:
Hmm now that I think about it, there is no purpose in having these virtual classes since in TypeScript, we always know if the `super.HOOK_NAME` exists.

Reply:
So based on our conversation, we should go about this in two ways:
1. Remove the abstract classes and just use an interface. This has the con of requiring a 'breaking change' label in the future if the user does not call super.
2. We use the workaround (compromise): https://github.com/microsoft/TypeScript/issues/21388 that enables us to change things internally without worrying about breaking changes for others.